acs-apiclient/LoginController.cs

```diff
@@ -36,23 +45,45 @@ public static LoginController Instance
                 return instance;
             }
         }
-        
+
+        /// <summary>
+        /// Destroys the instance.
+        /// </summary>
         public static void DestroyInstance()
         {
             instance = null;
         }
-        
+
+        /// <summary>
+        /// Setup the specified configuration and loginDelegate.
+        /// </summary>
+        /// <param name="configuration">Configuration.</param>
+        /// <param name="loginDelegate">Login delegate.</param>
         public void Configure(LoginConfiguration configuration, LoginDelegate loginDelegate)
         {
             this.configuration = configuration;
             this.loginDelegate = loginDelegate;
         }
 
         /// <summary>
-        /// Gets and sets the selected identity provider.
+        /// Initializes an ApiClient with a token string if it is valid.
         /// </summary>
-        /// <value>The selected identity provider.</value>
-        internal IdentityProvider selectedIdentityProvider { get; private set; }
+        /// <param name="tokenString">Token string.</param>
+        /// <returns><c>true</c>, if the token is valid, <c>false</c> otherwise.</returns>
+        public bool InitializesWithTokenString(string tokenString)
+        {
+            if (Utility.ValidateToken(tokenString))
+            {
+                var tokens = tokenString.Split('.');
+                MakeApiClient(
+                    Utility.Base64Decode(tokens[0]),
+                    Utility.Base64Decode(tokens[1]),
+                    Utility.Base64Decode(tokens[2])
```

Comment:
Consider creating constant for '0', '1', '2'.

Reply:
👍

Reply:
:white_check_mark: